### DISCUSSION TITLE
Use Verdaccio as a proxy cache.

### DISCUSSION THREAD

**Discussion by droslean** [verdaccio/verdaccio#3805] [Q&A]
Is it possible to configure Verdaccio to be used as a proxy cache only? 

If this is supported, can someone point out the documentation to achieve this?

**Comment by juanpicado** [comment]
> Is it possible to configure Verdaccio to be used as a proxy cache only?

What do you mean with only? What else it does you don't wan to do. Could you elaborate?

**Comment by droslean** [comment]
Currently, `yarn.lock` needs to point to the Verdaccio registry. I was wondering if it is possible to configure yarn/npm to use Verdaccio as a proxy only.

**Reply by juanpicado** [reply]
Do you mean yarn v1.x ?

**Reply by droslean** [reply]
Yes. I mean `yarn config set proxy` and `yarn config set https-proxy`.

**Reply by sacru2red** [reply]
I'm not entirely sure if I understand correctly, do you need to use a command like `yarn config set proxy`?

It seems that setting it in the project's configuration files such as .npmrc (for npm) or .yarnrc (for Yarn Classic) or yarnrc.yml (for Yarn Berry) should be sufficient to meet your use case.

For example, you can create an .npmrc file in a new project and add registry=https://verdaccio.mydomain.com, then try running `npm install somePackageInMyRegistry --verbose` to see if it works.

**Reply by droslean** [reply]
Using Verdaccio as registry forces all projects to use yarn.lock file that points to the Verdaccio domain instead. For now, our team unblocked that by manually replacing all registries to the Verdaccio registry in the yarn.lock file using `sed` commands... This is really bad. So before stop using Verdaccio at all, I was wondering if there is an option to use Verdaccio as a proxy cache only instead.

**Reply by juanpicado** [reply]
Have you read about the usage of `.npmrc` file ? https://verdaccio.org/docs/setup-npm#specific

**Reply by droslean** [reply]
@juanpicado Coming back to this. Thanks for the response but this is not an answer. I don't want to use verdaccio as a registry, but only as a proxy cache. The document you shared, will use the verdaccio as a registry instead which means that the links of all dependencies will point to that registry instead.

**Reply by droslean** [reply]
Here was our workaround: replacing the yarnpkg registry with the internal Verdaccio one. 
```sh
if [ -n "${VERDACCIO_REGISTRY}" ]; then
    sed -i "s|resolved \"https://registry.yarnpkg.com|resolved \"${VERDACCIO_REGISTRY%%/}|g" yarn.lock
    yarn config set registry "${VERDACCIO_REGISTRY}"
    yarn config set "strict-ssl" false
fi
```

To my understanding, what I am asking in this issue is not supported by Verdaccio. My team stopped using Verdaccio for now and moved to https://github.com/runk/npm-proxy-cache. If this feature will be implemented in the future, I would be happy to return to Verdaccio.
